runtime/vm/stringhelpers.cpp

```diff
@@ -258,179 +258,95 @@ compareStringToUTF8(J9VMThread *vmThread, j9object_t string, UDATA translateDots
 }
 
 #if !defined (J9VM_OUT_OF_PROCESS)
-
 /**
- * Copy certain number of Unicode characters from an offset into a Unicode character array to a UTF8 data buffer.
+ * Copy a Unicode String to a UTF8 data buffer with NULL termination. This function makes an assumption that the
+ * caller has guaranteed the buffer is of the appropriate size to fit the encoded UTF8 string.
  *
  * @param[in] vmThread the current J9VMThread
- * @param[in] compressed if the Unicode character array is compressed
- * @param[in] nullTermination if the utf8 data is going to be NULL terminated
+ * @param[in] string a string object to be copied, it can't be NULL
  * @param[in] stringFlags the flag to determine performing '.' --> '/'
- * @param[in] unicodeBytes a Unicode character array
- * @param[in] unicodeOffset an offset into the Unicode character array
- * @param[in] unicodeLength the number of Unicode characters to be copied
  * @param[in] utf8Data a utf8 data buffer
- * @param[in] utf8Length the size of the utf8 data buffer
  *
- * @return UDATA_MAX if a failure occurred, otherwise the number of utf8 data copied excluding null termination
+ * @return The computed length (in bytes) of the copied UTF8 string in the buffer excluding the NULL terminator.
```

Comment:
Pls remove the comment above, it is only required in the header file `vm_api.h`.

Reply:
Pls rebase for next commit.

Reply:
Fixed in 1baa27d